onnxruntime/python/ort_trainer.py

```diff
@@ -275,15 +275,20 @@ def convert_model_loss_fn_to_onnx(model, loss_fn, model_desc, device):
     input_names = [input.name_ for input in model_desc.inputs_]
     output_names = [output.name_ for output in model_desc.outputs_]
 
-    sample_inputs = []
-    for input_desc in model_desc.inputs_:
-        input_sample = generate_sample(input_desc, device)
-        sample_inputs.append(input_sample)
-
-    sample_outputs = []
-    for output_desc in model_desc.outputs_:
-        output_sample = generate_sample(output_desc, device)
-        sample_outputs.append(output_sample)
+    if isinstance(inputs, torch.Tensor):
+        inputs = [inputs]
+    if isinstance(inputs, dict):
+        sample_inputs = [inputs[k.name_].to(device=device) for k in model_desc.inputs_]
```

Comment:
What should happen if `inputs` and model_desc.inputs_ doesn't have the same keys?

Reply:
there is an assumption that names in model_desc must match model's arguments. maybe we shall discuss is there a better way to relax this assumption. Otherwise, we need to say this clearly in ORTTrainer API.

Reply:
For the happy path, this assumption works fine. But as `inputs` and `model_desc` are 2 different input parameters that come from different places, users could easily make a mistake by accident.

To make this check, we could do something like:
`inputs_matches = not (set(inputs) - set(model_desc.inputs_)) and not (set(model_desc.inputs_) - set(inputs))`
`input_matches` is `True` when all keys in `inputs` are present in `model_desc.inputs_` and vice-versa. Any different leads to False

What do you think? Depending on the size of the model, this check could take some time.

Reply:
`wrap_for_input_match` seems to do a similar check.

Reply:
There are actually three different input parameters. `inputs`, `model_desc` and `model.forward`.
`inputs`: type can be dict, thus inputs are not ordered.
`model_desc`: defines inputs by a list, inputs are ordered.
`model.forward`: specifies all inputs, inputs are ordered, however inputs may have default values.

This part can be further refactored. I'm not sure if it should be done in this PR.

Reply:
I am OK doing this down the road